xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/Registration.xml

```diff
@@ -192,6 +192,10 @@
         'mandatory' : {
           'failureMessage' : $services.localization.render('core.validation.required.message')
         },
+        'regex' : {
+          'pattern' : '/^[a-zA-Z0-9_]+$/',
```

Comment:
You're forcing the default value from ``createUser`` but this doesn't mean that you remove the inconsistency: you still have it if the user configures a different value for the ``xwiki.validusername`` property.

Reply:
Indeed, so if I understand correctly we need to introduce a `getValidUserNameRegEx()` somewhere as you suggest. I'm wondering what could be the best place. Could it be in `DefaultUserConfiguration` and `UserScriptService` maybe?